mailbox/store/src/main/java/org/apache/james/mailbox/store/ImmutableMailboxMessage.java

```diff
@@ -88,23 +88,19 @@ public ImmutableMailboxMessage from(MailboxId mailboxId, MailboxMessage message)
             }
             return ImmutableList.of();
         }
-
-        private static SharedByteArrayInputStream copy(InputStream inputStream) throws IOException {
-            return new SharedByteArrayInputStream(IOUtils.toByteArray(inputStream));
-        }
     }
 
     private final MessageId messageId;
     private final Date internalDate;
-    private final InputStream bodyContent;
+    private final byte[] bodyContent;
     private final String mediaType;
     private final String subType;
     private final long bodyOctets;
     private final long fullContentOctets;
     private final long headerOctets;
     private final Long textualLineCount;
-    private final InputStream headerContent;
-    private final InputStream fullContent;
+    private final byte[] headerContent;
+    private final byte[] fullContent;
```

Comment:
That means we cary the double of what is needed, isn't it? (headerContent + bodyContent + fullContent)

Can't we manage to have something with an offset?

Reply:
Maybe something like _Bytes.concat_
but I would like to have Jenkins feedback on this pull request before optimizing things.